core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -605,6 +603,22 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  def unZip(inFile: File, untarDir: File): Unit = {
+    val untarCommand = new StringBuffer
+    untarCommand.append(s"mkdir $untarDir && unzip -d ")
+    untarCommand.append(FileUtil.makeSecureShellPath(untarDir))
+    untarCommand.append(" -o ")
+    untarCommand.append(FileUtil.makeSecureShellPath(inFile))
+
+    val shellCmd = Array("bash", "-c", untarCommand.toString)
```

Comment:
The problem here is that it won't support Windows. It's generally not a good practice to rely on external problems that has to be manually installed. If `unzip` is not installed, it will just fail.

Reply:
Yes, the unzip method provided by the previous fileutil by the window operating system can retain the permissions of the original file, so I can use the judgment statement to execute the shell command when the Linux operating system is found, and the window operating system retains the original method.

Reply:
In fact, this way of relying on external installation also draws lessons from the decompression method of untar.

Reply:
@HyukjinKwon I have revised it again. Could you review it.

Reply:
I think untar uses Java library not the external ones. I still think we should avoid relying on external programs. Users can use tar.gz if they want to keep file permissions